### DISCUSSION TITLE
Use OpenAMP to Communicate between several Hypervisors

### DISCUSSION THREAD

**Discussion by yannik93** [OpenAMP/open-amp#268] [General]
Hi together, 

iam writing nowadays my thesis about using the OpenAMP Framework in the so called Vehicle Computers.
Like in the next generation of ECUs you will have architectures with multiple hypervisors and the communication in the hypervisors is already "standardized", but not between the hypervisors. I have attached two possible architectures, how it may could be possible to communicate between hypervisors. 
[Automotive_Vehicle_Computer_OpenAMP_Hypervisor.pdf](https://github.com/OpenAMP/open-amp/files/5958691/Automotive_Vehicle_Computer_OpenAMP_Hypervisor.pdf)
[Automotive_Vehicle_Computer_OpenAMP_Plain_5.pdf](https://github.com/OpenAMP/open-amp/files/5958753/Automotive_Vehicle_Computer_OpenAMP_Plain_5.pdf)

So actually my question is if someone knows if those (or even one) architectures could be even work and with what overhead it would work. What architecture would be the better option? Maybe someone already has done some work like that.

**Comment by edmooring** [comment]
Hi,

Sorry about the delay in responding. I think either architecture will work, but the first one, where AutoSAR is actually the OpenAMP master will be easier to implement. If the hypervisor(s) you are using support some form of device pass-through, the overhead could be very small.

Any more detailed answer depends on:

- The hardware architecture
- Whether the RTOS on which AutoSAR is running supports OpenAMP
- The hypervisor(s) chosen.

**Reply by yannik93** [reply]
Hi,
first of all thanks for your response. 
I dont have any specification on the hypervisor or the hw. 
I just have a look how OpenAMP could be applied to such SW-Architectures in a good manner. 
I also thought the first concept ist the best and easisest toi handle, but i dont know how i could realize the start of the OS. Because the Hypervisor at that domain needs to be started earlier. In the second concept i could start the hypervisor via the Master and the hypervisor could handle the start of his virtual guests.
